### DISCUSSION TITLE
Workflow 3.0 - Error in Designer when Creating Workflow with Custom Activity

### DISCUSSION THREAD

**Discussion by vsanghavi-conga** [elsa-workflows/elsa-core#3738] [Q&A]
Hi,

I have created custom activity with one parameter in its constructor instance. When I run application designer shows this activity and I can drag it on designer as well.

![image](https://user-images.githubusercontent.com/122509094/221159619-9530e8b6-d5c3-4331-b7c4-68c2c86ba895.png)

But when I try to Save workflow I get this error - Each parameter in the deserialization constructor on type 'WorkflowServer.Api.CustomActivities.HelloWorldActivity' must bind to an object property or field on deserialization. Each parameter name must match with a property or field on the object. Fields are only considered when 'JsonSerializerOptions.IncludeFields' is enabled. The match can be case-insensitive.
.
![image](https://user-images.githubusercontent.com/122509094/221159305-ecbe8ada-3c6b-4468-ab71-8e2c3f4842f8.png)

It works fine if Custom Activity has constructor with no arguments.

Regards,
Viraj

**Comment by sfmskywalker** [comment]
Hi @vsanghavi-conga 

In Elsa 3, activity classes should have at least one default constructor.
In order to invoke services, you need to resolve them from the `Execute` / `ExecuteAsync` method. For example:

```csharp
public class HelloWorldActivity : Activity
{

   public HelloWorldActivity()
   {
      Id = "HelloWorldActivity"; // Don't actually do this - activity IDs need to be unique within the same workflow. If you give a fixed ID, you can use it only once in a given workflow.
      CanStartWorkflow = true; // Better to don't do this from within the activity, but instead, from the workflow. Otherwise, all activitiy instances of this type will have this value set, which may not be what you want (unless you do)
   }

   public override Execute(ActivityExecutionContext context)
   {
      var logger = context.GetRequiredService<ILogger<HelloWorldActivity>>();

      logger.LogDebug("Hello world!");
   }
}
```
